File: bug70_wrong_iteration_step/bug70_wrong_iteration_step/main1.js
function main() {
    let data = [];
    for (let i = 0n; i < 10n; i++) data.push(0xBBBBn * (i + 1n));

    let acc = 0n;

    // FIX: changed j += 2 → j++ (was skipping elements)
    for (let j = 0; j < 10; j++) {
        let i = BigInt(j);
        acc ^= data[j] * (i + 1n);
        acc = (acc * 0x9E3779B97F4A7C15n) & 0xFFFFFFFFFFFFFFFFn;
        acc = ((acc << (i % 11n + 1n)) | (acc >> (63n - i % 11n))) & 0xFFFFFFFFFFFFFFFFn;
    }

    // FIX: print result
    console.log("Result:", acc.toString(16));
}

main();

File: bug70_wrong_iteration_step/bug70_wrong_iteration_step/main1.java
public class main1 {
    public static void main(String[] args) {
        long acc = 0;
        long[] data = new long[20];

        for (int i = 0; i < 20; i++)
            data[i] = i;

        // FIX: step = 2
        for (int i = 0; i < 20; i += 2) {
            acc ^= data[i] * 0x9E3779B97F4A7C15L;
            acc &= 0xFFFFFFFFFFFFFFFFL;
            acc = ((acc << (i % 9 + 1)) | (acc >>> (63 - i % 9))) & 0xFFFFFFFFFFFFFFFFL;
        }

        String result = "0x" + Long.toHexString(acc);
        System.out.println(result);
    }
}
